### ISSUE TITLE
It can't work.

### ISSUE THREAD

**Issue by Zibri** [kushpr/TTL_Changer#2]
This solution can't work because there is no TTL-SET in iptables on android.

`iptables v1.8.7 (legacy): unknown option "--ttl-set"
`

moreover, setting TTL to 65 might work but is incorrect. TTL should be just increaded by 1.

My solution:
I patch the kernel so TTL is not decreased when forwarding.
As of now only a few phones are supported, but it works perfectly:
https://github.com/Zibri/ttl_fix/releases/tag/V3

**Comment by kushpr** [comment]
As you said most phones require a patched or custom kernel for this to work.
Our kernels do bring the option for ttl-set but by factory its on N. So
again all devices are supported just need a custom or patched kernel.

**Comment by Zibri** [comment]
> As you said most phones require a patched or custom kernel for this to work. Our kernels do bring the option for ttl-set but by factory its on N. So again all devices are supported just need a custom or patched kernel.

Even with a custom kernel, is preferred to remove the TTL decrease from ip_forward.c than leave it and then add 1 to incoming TTL... imho.

**Comment by kushpr** [comment]
Correct. Theres actually a few ways to go about it you could edit the
kernel modules or simply just go tru iptable rules or netfilters theres so
many ways to do this workaround
